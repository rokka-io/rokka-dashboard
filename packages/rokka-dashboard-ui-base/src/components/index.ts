export * from './formGroup/formGroup';
export * from './loadingIndicatingButton/loadingIndicatingButton';
export * from './signForm/signForm';
export * from './user/user';
