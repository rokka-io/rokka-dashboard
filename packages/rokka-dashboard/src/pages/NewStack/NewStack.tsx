import React, { FunctionComponent } from 'react';

interface NewStackProps {}

export const NewStack: FunctionComponent<NewStackProps> = () => {
  return <div>NewStack</div>;
};
