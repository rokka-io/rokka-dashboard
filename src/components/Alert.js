import PropTypes from 'prop-types'
import React from 'react'
import cx from 'classnames'

const Alert = ({ alert = null }) => {
  if (!alert) {
    return null
  }

  let message = alert.message
  if (Array.isArray(message)) {
    message = message.map((msg, index) => (<span className="rka-alert-msg" key={index}>{msg}</span>))
  } else {
    message = (<span className="rka-alert-msg">{message}</span>)
  }

  return <div className={cx('rka-alert', `is-${alert.type}`, 'txt-c')}>{message}</div>
}
Alert.propTypes = {
  alert: PropTypes.shape({
    type: PropTypes.string,
    message: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)])
  })
}

export default Alert
