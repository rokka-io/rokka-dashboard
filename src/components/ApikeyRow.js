import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import rokka from '../rokka'
import { formatDate } from '../utils/string'

const DEFAULT_STATE = {
  currentKeyId: '',
  delete: false
}

class ApikeyRow extends PureComponent {
  constructor(props) {
    super(props)
    this.state = DEFAULT_STATE
  }
  showDelete = () => {
    this.setState({ delete: true })
  }

  reallyDelete = () => {
    rokka()
      .user.deleteApiKey(this.props.apiKey.id)
      .then(({ body }) => {
        this.props.updateKeys()
      })
      .catch(err => {
        alert("Api Key deletion didn't work:" + err.body.error.message)
      })
  }
  render() {
    return (
      <tr key={this.props.apiKey.id}>
        <td className={'mb-md'} style={{ height: '4em' }}>
          {this.props.apiKey.id}
        </td>
        <td>{this.props.apiKey.comment}</td>
        <td>{formatDate(this.props.apiKey.created, 'Before December 2021')}</td>
        <td>{formatDate(this.props.apiKey.accessed, 'Before December 2021')}</td>
        <td>
          {this.props.apiKey.id === this.props.currentKeyId ? (
            <>
              <div className={'mb-sm'}>Currently used Api Key, not deletable </div>
              <div>(use another Api Key to login, if you want to delete this one)</div>
            </>
          ) : this.state.delete ? (
            <>
              <button
                className="rka-button rka-button-brand  mr-md"
                onClick={e => this.reallyDelete(e)}
              >
                Really Delete!
              </button>
              <button
                className="rka-button rka-button-secondary"
                onClick={e => this.setState({ delete: false })}
              >
                Cancel
              </button>
            </>
          ) : (
            <button className="rka-button rka-button-brand" onClick={e => this.showDelete(e)}>
              Delete Api Key
            </button>
          )}
        </td>
      </tr>
    )
  }
}

ApikeyRow.propTypes = {
  apiKey: PropTypes.object.isRequired,
  currentKeyId: PropTypes.string.isRequired,
  updateKeys: PropTypes.func.isRequired
}

export default ApikeyRow
