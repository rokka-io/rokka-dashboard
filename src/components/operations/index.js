import PropTypes from 'prop-types'
import React from 'react'
import { findDOMNode } from 'react-dom'
import { DragSource, DropTarget } from 'react-dnd'
import cx from 'classnames'
import factory from './factory'

const ITEM_TYPE = 'OPERATION'

const Operation = ({
  availableOperations,
  operation,
  index,
  isActive,
  onChange,
  removeOperation,
  setActiveOperation,
  isDragging,
  isOver,
  connectDragSource,
  connectDragPreview,
  connectDropTarget
}) => {
  const hasErrors = Object.keys(operation.errors).length
  const className = cx('pa-md', 'mb-xs', 'pos-r', {
    'bg-gray-lightest': index % 2,
    'is-active': isActive || hasErrors,
    'bor-cranberry': hasErrors,
    'bor-light': !hasErrors,
    'opaque': isDragging,
    'bor-brand-light-dashed': isOver
  })

  return connectDragPreview(connectDropTarget(
    <div className={className} key={`operations-${index}`}>
      {connectDragSource(
        <div>
          <a className="rka-close-icon" href="#" onClick={e => removeOperation(e, index)} title="Remove" />
          <a className="rka-edit-icon" href="#" onClick={e => setActiveOperation(e, index)} />
          <a className="rka-move-icon" href="#" onClick={e => e.preventDefault()} />
          <h3 className="rka-h3 txt-cap">{operation.name}</h3>
        </div>
      )}
      <div className={cx({'semi-transparent': isOver})}>
        <div className="rka-operation-content">
          {factory(availableOperations, operation.name, operation.options, (e) => onChange(index, e), operation.errors)}
        </div>
      </div>
    </div>
  ))
}
Operation.propTypes = {
  availableOperations: PropTypes.object.isRequired,
  operation: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    options: PropTypes.object.isRequired,
    errors: PropTypes.object.isRequired
  }).isRequired,
  index: PropTypes.number.isRequired,
  isActive: PropTypes.bool.isRequired,
  // state manipulators on container
  onChange: PropTypes.func.isRequired,
  onMoveOperation: PropTypes.func.isRequired,
  removeOperation: PropTypes.func.isRequired,
  setActiveOperation: PropTypes.func.isRequired,
  // from HOCs of react-dnd
  connectDragSource: PropTypes.func.isRequired,
  connectDragPreview: PropTypes.func.isRequired,
  connectDropTarget: PropTypes.func.isRequired,
  isDragging: PropTypes.bool.isRequired,
  isOver: PropTypes.bool.isRequired
}

const dragSourceSpec = {
  beginDrag (props) {
    return {
      id: props.operation.id,
      index: props.index
    }
  },

  endDrag (props, monitor) {
    const { id: droppedId, originalIndex } = monitor.getItem()
    const didDrop = monitor.didDrop()

    if (!didDrop) {
      props.onMoveOperation(droppedId, originalIndex)
    }
  }
}

const dragSourceCollect = (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  connectDragPreview: connect.dragPreview(),
  isDragging: monitor.isDragging()
})

const dropTargetSpec = {
  hover (props, monitor, component) {
    const dragIndex = monitor.getItem().index
    const hoverIndex = props.index

    // Don't replace items with themselves
    if (dragIndex === hoverIndex) {
      return
    }

    // Determine rectangle on screen
    const hoverBoundingRect = findDOMNode(component).getBoundingClientRect()

    // Get vertical middle
    const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2

    // Determine mouse position
    const clientOffset = monitor.getClientOffset()

    // Get pixels to the top
    const hoverClientY = clientOffset.y - hoverBoundingRect.top

    // Only perform the move when the mouse has crossed half of the items height
    // When dragging downwards, only move when the cursor is below 50%
    // When dragging upwards, only move when the cursor is above 50%

    // Dragging downwards
    if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
      return
    }

    // Dragging upwards
    if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
      return
    }

    // Time to actually perform the action
    props.onMoveOperation(dragIndex, hoverIndex)

    // Note: we're mutating the monitor item here!
    // Generally it's better to avoid mutations,
    // but it's good here for the sake of performance
    // to avoid expensive index searches.
    monitor.getItem().index = hoverIndex
  }
}

const dropTargetCollect = (connect, monitor) => ({
  connectDropTarget: connect.dropTarget(),
  isOver: monitor.isOver()
})

export default DropTarget(ITEM_TYPE, dropTargetSpec, dropTargetCollect)(DragSource(ITEM_TYPE, dragSourceSpec, dragSourceCollect)(Operation))
