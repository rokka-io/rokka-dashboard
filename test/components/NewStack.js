import React, { Component } from 'react'
import renderer from 'react-test-renderer'
import { DragDropContext } from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'
import { NewStack } from '../../src/components/NewStack'
import operations from '../operations.json'
import stackOptions from '../stackoptions.json'
import cloneStack from '../cloneStack.json'

// N.B. this test currently triggers a warning because
//      onComponentDidMount calls an update on the state.
//      Making this better testable is one of the goals of refactoring.

function wrapInTestContext (DecoratedComponent) {
  class TestContextContainer extends Component {
    render () {
      return <DecoratedComponent {...this.props} />
    }
  }

  return DragDropContext(TestBackend)(TestContextContainer)
}

test('NewStack does render with minimal props', () => {
  const auth = {
    organization: 'test-org'
  }
  const router = {
    push: () => {
      throw new Error('should not be called')
    }
  }
  const onOpenChoosePreviewImage = () => {
    throw new Error('should not be called')
  }
  const loadPreviewImage = () => {}

  const props = {
    auth,
    operations,
    router,
    onOpenChoosePreviewImage,
    loadPreviewImage
  }

  const component = renderer.create(
    <NewStack {...props} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NewStack does render with a previewImage', () => {
  const auth = {
    organization: 'test-org'
  }
  const router = {
    push: () => {
      throw new Error('should not be called')
    }
  }
  const onOpenChoosePreviewImage = () => {
    throw new Error('should not be called')
  }
  const loadPreviewImage = () => {}
  const previewImage = {
    hash: '7cb0101d53f1579c0e0da11f2407146eebeca638',
    format: 'image/png'
  }

  const props = {
    auth,
    operations,
    router,
    onOpenChoosePreviewImage,
    loadPreviewImage,
    previewImage
  }

  const component = renderer.create(
    <NewStack {...props} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NewStack does render with stackOptions', () => {
  const auth = {
    organization: 'test-org'
  }
  const router = {
    push: () => {
      throw new Error('should not be called')
    }
  }
  const onOpenChoosePreviewImage = () => {
    throw new Error('should not be called')
  }
  const loadPreviewImage = () => {}
  const previewImage = {
    hash: '7cb0101d53f1579c0e0da11f2407146eebeca638',
    format: 'image/png'
  }

  const props = {
    auth,
    operations,
    router,
    onOpenChoosePreviewImage,
    loadPreviewImage,
    previewImage,
    stackOptions
  }

  const component = renderer.create(
    <NewStack {...props} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('NewStack does render with stacks', () => {
  const auth = {
    organization: 'test-org'
  }
  const router = {
    push: () => {
      throw new Error('should not be called')
    }
  }
  const onOpenChoosePreviewImage = () => {
    throw new Error('should not be called')
  }
  const loadPreviewImage = () => {}
  const previewImage = {
    hash: '7cb0101d53f1579c0e0da11f2407146eebeca638',
    format: 'image/png'
  }
  const stacks = {
    items: [
      {name: 'stack-a'},
      {name: 'stack-b'},
      {name: 'stack-c'}
    ]
  }

  const props = {
    auth,
    operations,
    router,
    onOpenChoosePreviewImage,
    loadPreviewImage,
    previewImage,
    stackOptions,
    stacks
  }

  const component = renderer.create(
    <NewStack {...props} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('newStack does render with clone stack props', () => {
  const auth = {
    organization: 'test-org'
  }
  const router = {
    push: () => {
      throw new Error('should not be called')
    }
  }
  const stackClone = cloneStack
  const onOpenChoosePreviewImage = () => {
    throw new Error('should not be called')
  }
  const loadPreviewImage = () => {}

  const props = {
    auth,
    operations,
    stackClone,
    router,
    onOpenChoosePreviewImage,
    loadPreviewImage
  }

  const WrappedNewStack = wrapInTestContext(NewStack)
  const component = renderer.create(
    <WrappedNewStack {...props} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
