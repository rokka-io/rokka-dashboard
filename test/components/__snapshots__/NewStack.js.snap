// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewStack does render with a previewImage 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="bg-white pa-md clearfix"
  >
    <h1
      className="rka-h1 flo-l mt-xs"
    >
      New stack
    </h1>
    <div
      className="flo-r"
    >
      <button
        className="rka-button rka-button-secondary mr-md"
        disabled={true}
        href="#"
        onClick={[Function]}
        type="button"
      >
        Update preview
      </button>
      <button
        className="rka-button rka-button-brand disabled flo-r"
        type="submit"
      >
        Create stack
      </button>
    </div>
  </div>
  <section
    className="rka-box rka-box-stacks pt-n"
  >
    <div
      className="row"
    >
      <div
        className="col-md-7 col-sm-7"
      >
        <h3
          className="rka-h2 mv-md"
        >
          Stack details
        </h3>
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Name
            <span>
               *
            </span>
          </label>
          <input
            className="rka-input-txt"
            id="name"
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <h3
          className="rka-h2 mv-md"
        >
          Operations
        </h3>
        <div
          className="pa-md bor-light mt-md"
        >
          <h3
            className="rka-h3 mb-md"
          >
            New operation
          </h3>
          <div
            className="rka-form-group"
          >
            <select
              className="rka-select"
            >
              <option
                value="alpha"
              >
                alpha
              </option>
              <option
                value="autorotate"
              >
                autorotate
              </option>
              <option
                value="blur"
              >
                blur
              </option>
              <option
                value="composition"
              >
                composition
              </option>
              <option
                value="crop"
              >
                crop
              </option>
              <option
                value="dropshadow"
              >
                dropshadow
              </option>
              <option
                value="grayscale"
              >
                grayscale
              </option>
              <option
                value="resize"
              >
                resize
              </option>
              <option
                value="rotate"
              >
                rotate
              </option>
              <option
                value="sepia"
              >
                sepia
              </option>
              <option
                value="trim"
              >
                trim
              </option>
            </select>
          </div>
          <a
            className="rka-button rka-button-brand rka-button-sm"
            href="#"
            onClick={[Function]}
          >
            Add operation
          </a>
        </div>
      </div>
      <div
        className="col-md-5 col-sm-5"
      >
        <h3
          className="rka-h2 mv-md"
        >
          Preview
          <a
            className="rka-link flo-r txt-sm"
            href="#"
            onClick={[Function]}
          >
            Change picture
          </a>
        </h3>
        <div
          className="rka-stack-img-container bg-chess mb-xs bor-light txt-c"
        >
          <p
            className="pa-md bg-white txt-l"
          >
            Customized 
            <a
              className="rka-link flo-r"
              href="https://test-org.rokka.io/dynamic/noop/7cb0101d53f1579c0e0da11f2407146eebeca638.image/png"
              target="_blank"
            >
              Open in new window
            </a>
          </p>
          <img
            src="https://test-org.rokka.io/dynamic/noop/7cb0101d53f1579c0e0da11f2407146eebeca638.image/png"
          />
        </div>
        <div
          className="rka-stack-img-container bg-chess bor-light txt-c"
        >
          <p
            className="pa-md bg-white txt-l"
          >
            Original 
            <a
              className="rka-link flo-r"
              href="https://test-org.rokka.io/dynamic/noop/7cb0101d53f1579c0e0da11f2407146eebeca638.image/png"
              target="_blank"
            >
              Open in new window
            </a>
          </p>
          <img
            src="https://test-org.rokka.io/dynamic/noop/7cb0101d53f1579c0e0da11f2407146eebeca638.image/png"
          />
        </div>
      </div>
    </div>
  </section>
</form>
`;

exports[`NewStack does render with minimal props 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="bg-white pa-md clearfix"
  >
    <h1
      className="rka-h1 flo-l mt-xs"
    >
      New stack
    </h1>
    <div
      className="flo-r"
    >
      <button
        className="rka-button rka-button-brand disabled flo-r"
        type="submit"
      >
        Create stack
      </button>
    </div>
  </div>
  <section
    className="rka-box rka-box-stacks pt-n"
  >
    <div
      className="row"
    >
      <div
        className="col-md-7 col-sm-7"
      >
        <h3
          className="rka-h2 mv-md"
        >
          Stack details
        </h3>
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Name
            <span>
               *
            </span>
          </label>
          <input
            className="rka-input-txt"
            id="name"
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <h3
          className="rka-h2 mv-md"
        >
          Operations
        </h3>
        <div
          className="pa-md bor-light mt-md"
        >
          <h3
            className="rka-h3 mb-md"
          >
            New operation
          </h3>
          <div
            className="rka-form-group"
          >
            <select
              className="rka-select"
            >
              <option
                value="alpha"
              >
                alpha
              </option>
              <option
                value="autorotate"
              >
                autorotate
              </option>
              <option
                value="blur"
              >
                blur
              </option>
              <option
                value="composition"
              >
                composition
              </option>
              <option
                value="crop"
              >
                crop
              </option>
              <option
                value="dropshadow"
              >
                dropshadow
              </option>
              <option
                value="grayscale"
              >
                grayscale
              </option>
              <option
                value="resize"
              >
                resize
              </option>
              <option
                value="rotate"
              >
                rotate
              </option>
              <option
                value="sepia"
              >
                sepia
              </option>
              <option
                value="trim"
              >
                trim
              </option>
            </select>
          </div>
          <a
            className="rka-button rka-button-brand rka-button-sm"
            href="#"
            onClick={[Function]}
          >
            Add operation
          </a>
        </div>
      </div>
    </div>
  </section>
</form>
`;

exports[`NewStack does render with stackOptions 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="bg-white pa-md clearfix"
  >
    <h1
      className="rka-h1 flo-l mt-xs"
    >
      New stack
    </h1>
    <div
      className="flo-r"
    >
      <button
        className="rka-button rka-button-secondary mr-md"
        disabled={true}
        href="#"
        onClick={[Function]}
        type="button"
      >
        Update preview
      </button>
      <button
        className="rka-button rka-button-brand disabled flo-r"
        type="submit"
      >
        Create stack
      </button>
    </div>
  </div>
  <section
    className="rka-box rka-box-stacks pt-n"
  >
    <div
      className="row"
    >
      <div
        className="col-md-7 col-sm-7"
      >
        <h3
          className="rka-h2 mv-md"
        >
          Stack details
        </h3>
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Name
            <span>
               *
            </span>
          </label>
          <input
            className="rka-input-txt"
            id="name"
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div>
          <h3
            className="rka-h2 mv-md"
          >
            Options
          </h3>
          <div
            className="bg-gray-lightest pa-md bor-light"
          >
            <div
              className="row"
            >
              <div
                className="col-md-6"
              >
                <div
                  className="rka-form-group"
                >
                  <label
                    className="rka-label"
                  >
                    PNG Image compression
                  </label>
                  <div
                    className="rka-input-range-container clearfix"
                  >
                    <input
                      className="rka-input-range"
                      disabled={false}
                      max={9}
                      min={0}
                      name="png.compression_level"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder={undefined}
                      type="range"
                      value={7}
                    />
                    <p>
                      <span
                        className="rka-input-range-min"
                      >
                        0
                      </span>
                      <input
                        className="rka-input-txt rka-input-range-current txt-c"
                        disabled={false}
                        name="png.compression_level"
                        onChange={[Function]}
                        placeholder={undefined}
                        size={5}
                        type="text"
                        value={7}
                      />
                      <span
                        className="rka-input-range-max"
                      >
                        9
                      </span>
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="rka-form-group"
                >
                  <label
                    className="rka-label"
                  >
                    JPG Image quality
                  </label>
                  <div
                    className="rka-input-range-container clearfix"
                  >
                    <input
                      className="rka-input-range"
                      disabled={false}
                      max={100}
                      min={1}
                      name="jpg.quality"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder={undefined}
                      type="range"
                      value={76}
                    />
                    <p>
                      <span
                        className="rka-input-range-min"
                      >
                        1
                      </span>
                      <input
                        className="rka-input-txt rka-input-range-current txt-c"
                        disabled={false}
                        name="jpg.quality"
                        onChange={[Function]}
                        placeholder={undefined}
                        size={5}
                        type="text"
                        value={76}
                      />
                      <span
                        className="rka-input-range-max"
                      >
                        100
                      </span>
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="rka-form-group"
                >
                  <label
                    className="rka-label"
                  >
                    WebP Image quality
                  </label>
                  <div
                    className="rka-input-range-container clearfix"
                  >
                    <input
                      className="rka-input-range"
                      disabled={false}
                      max={100}
                      min={1}
                      name="webp.quality"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder={undefined}
                      type="range"
                      value={80}
                    />
                    <p>
                      <span
                        className="rka-input-range-min"
                      >
                        1
                      </span>
                      <input
                        className="rka-input-txt rka-input-range-current txt-c"
                        disabled={false}
                        name="webp.quality"
                        onChange={[Function]}
                        placeholder={undefined}
                        size={5}
                        type="text"
                        value={80}
                      />
                      <span
                        className="rka-input-range-max"
                      >
                        100
                      </span>
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="col-md-6"
              />
              <div
                className="col-md-12"
              />
            </div>
          </div>
        </div>
        <h3
          className="rka-h2 mv-md"
        >
          Operations
        </h3>
        <div
          className="pa-md bor-light mt-md"
        >
          <h3
            className="rka-h3 mb-md"
          >
            New operation
          </h3>
          <div
            className="rka-form-group"
          >
            <select
              className="rka-select"
            >
              <option
                value="alpha"
              >
                alpha
              </option>
              <option
                value="autorotate"
              >
                autorotate
              </option>
              <option
                value="blur"
              >
                blur
              </option>
              <option
                value="composition"
              >
                composition
              </option>
              <option
                value="crop"
              >
                crop
              </option>
              <option
                value="dropshadow"
              >
                dropshadow
              </option>
              <option
                value="grayscale"
              >
                grayscale
              </option>
              <option
                value="resize"
              >
                resize
              </option>
              <option
                value="rotate"
              >
                rotate
              </option>
              <option
                value="sepia"
              >
                sepia
              </option>
              <option
                value="trim"
              >
                trim
              </option>
            </select>
          </div>
          <a
            className="rka-button rka-button-brand rka-button-sm"
            href="#"
            onClick={[Function]}
          >
            Add operation
          </a>
        </div>
      </div>
      <div
        className="col-md-5 col-sm-5"
      >
        <h3
          className="rka-h2 mv-md"
        >
          Preview
          <a
            className="rka-link flo-r txt-sm"
            href="#"
            onClick={[Function]}
          >
            Change picture
          </a>
        </h3>
        <div
          className="rka-stack-img-container bg-chess mb-xs bor-light txt-c"
        >
          <p
            className="pa-md bg-white txt-l"
          >
            Customized 
            <a
              className="rka-link flo-r"
              href="https://test-org.rokka.io/dynamic/noop/7cb0101d53f1579c0e0da11f2407146eebeca638.image/png"
              target="_blank"
            >
              Open in new window
            </a>
          </p>
          <img
            src="https://test-org.rokka.io/dynamic/noop/7cb0101d53f1579c0e0da11f2407146eebeca638.image/png"
          />
        </div>
        <div
          className="rka-stack-img-container bg-chess bor-light txt-c"
        >
          <p
            className="pa-md bg-white txt-l"
          >
            Original 
            <a
              className="rka-link flo-r"
              href="https://test-org.rokka.io/dynamic/noop/7cb0101d53f1579c0e0da11f2407146eebeca638.image/png"
              target="_blank"
            >
              Open in new window
            </a>
          </p>
          <img
            src="https://test-org.rokka.io/dynamic/noop/7cb0101d53f1579c0e0da11f2407146eebeca638.image/png"
          />
        </div>
      </div>
    </div>
  </section>
</form>
`;

exports[`NewStack does render with stacks 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="bg-white pa-md clearfix"
  >
    <h1
      className="rka-h1 flo-l mt-xs"
    >
      New stack
    </h1>
    <div
      className="flo-r"
    >
      <button
        className="rka-button rka-button-secondary mr-md"
        disabled={true}
        href="#"
        onClick={[Function]}
        type="button"
      >
        Update preview
      </button>
      <button
        className="rka-button rka-button-brand disabled flo-r"
        type="submit"
      >
        Create stack
      </button>
    </div>
  </div>
  <section
    className="rka-box rka-box-stacks pt-n"
  >
    <div
      className="row"
    >
      <div
        className="col-md-7 col-sm-7"
      >
        <h3
          className="rka-h2 mv-md"
        >
          Stack details
        </h3>
        <div
          className="rka-form-group"
        >
          <label
            className="rka-label"
          >
            Name
            <span>
               *
            </span>
          </label>
          <input
            className="rka-input-txt"
            id="name"
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div>
          <h3
            className="rka-h2 mv-md"
          >
            Options
          </h3>
          <div
            className="bg-gray-lightest pa-md bor-light"
          >
            <div
              className="row"
            >
              <div
                className="col-md-6"
              >
                <div
                  className="rka-form-group"
                >
                  <label
                    className="rka-label"
                  >
                    PNG Image compression
                  </label>
                  <div
                    className="rka-input-range-container clearfix"
                  >
                    <input
                      className="rka-input-range"
                      disabled={false}
                      max={9}
                      min={0}
                      name="png.compression_level"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder={undefined}
                      type="range"
                      value={7}
                    />
                    <p>
                      <span
                        className="rka-input-range-min"
                      >
                        0
                      </span>
                      <input
                        className="rka-input-txt rka-input-range-current txt-c"
                        disabled={false}
                        name="png.compression_level"
                        onChange={[Function]}
                        placeholder={undefined}
                        size={5}
                        type="text"
                        value={7}
                      />
                      <span
                        className="rka-input-range-max"
                      >
                        9
                      </span>
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="rka-form-group"
                >
                  <label
                    className="rka-label"
                  >
                    JPG Image quality
                  </label>
                  <div
                    className="rka-input-range-container clearfix"
                  >
                    <input
                      className="rka-input-range"
                      disabled={false}
                      max={100}
                      min={1}
                      name="jpg.quality"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder={undefined}
                      type="range"
                      value={76}
                    />
                    <p>
                      <span
                        className="rka-input-range-min"
                      >
                        1
                      </span>
                      <input
                        className="rka-input-txt rka-input-range-current txt-c"
                        disabled={false}
                        name="jpg.quality"
                        onChange={[Function]}
                        placeholder={undefined}
                        size={5}
                        type="text"
                        value={76}
                      />
                      <span
                        className="rka-input-range-max"
                      >
                        100
                      </span>
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="col-md-6"
              >
                <div
                  className="rka-form-group"
                >
                  <label
                    className="rka-label"
                  >
                    WebP Image quality
                  </label>
                  <div
                    className="rka-input-range-container clearfix"
                  >
                    <input
                      className="rka-input-range"
                      disabled={false}
                      max={100}
                      min={1}
                      name="webp.quality"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder={undefined}
                      type="range"
                      value={80}
                    />
                    <p>
                      <span
                        className="rka-input-range-min"
                      >
                        1
                      </span>
                      <input
                        className="rka-input-txt rka-input-range-current txt-c"
                        disabled={false}
                        name="webp.quality"
                        onChange={[Function]}
                        placeholder={undefined}
                        size={5}
                        type="text"
                        value={80}
                      />
                      <span
                        className="rka-input-range-max"
                      >
                        100
                      </span>
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="col-md-6"
              />
              <div
                className="col-md-12"
              >
                <div
                  className="rka-form-group mb-n"
                >
                  <label
                    className="rka-label"
                  >
                    Base Stack
                  </label>
                  <div
                    className="Select Select--single is-clearable is-searchable"
                    style={undefined}
                  >
                    <div
                      className="Select-control"
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      style={undefined}
                    >
                      <span
                        className="Select-multi-value-wrapper"
                        id="react-select-2--value"
                      >
                        <div
                          className="Select-placeholder"
                        >
                          Enter base stack, default: no base stack used
                        </div>
                        <div
                          className="Select-input"
                          style={
                            Object {
                              "display": "inline-block",
                            }
                          }
                        >
                          <input
                            aria-activedescendant="react-select-2--value"
                            aria-describedby={undefined}
                            aria-expanded="false"
                            aria-haspopup="false"
                            aria-label={undefined}
                            aria-labelledby={undefined}
                            aria-owns=""
                            className={undefined}
                            id={undefined}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={false}
                            role="combobox"
                            style={
                              Object {
                                "boxSizing": "content-box",
                                "width": "5px",
                              }
                            }
                            tabIndex={undefined}
                            value=""
                          />
                          <div
                            style={
                              Object {
                                "height": 0,
                                "left": 0,
                                "overflow": "scroll",
                                "position": "absolute",
                                "top": 0,
                                "visibility": "hidden",
                                "whiteSpace": "pre",
                              }
                            }
                          >
                            
                          </div>
                        </div>
                      </span>
                      <span
                        className="Select-arrow-zone"
                        onMouseDown={[Function]}
                      >
                        <span
                          className="Select-arrow"
                          onMouseDown={[Function]}
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <h3
          className="rka-h2 mv-md"
        >
          Operations
        </h3>
        <div
          className="pa-md bor-light mt-md"
        >
          <h3
            className="rka-h3 mb-md"
          >
            New operation
          </h3>
          <div
            className="rka-form-group"
          >
            <select
              className="rka-select"
            >
              <option
                value="alpha"
              >
                alpha
              </option>
              <option
                value="autorotate"
              >
                autorotate
              </option>
              <option
                value="blur"
              >
                blur
              </option>
              <option
                value="composition"
              >
                composition
              </option>
              <option
                value="crop"
              >
                crop
              </option>
              <option
                value="dropshadow"
              >
                dropshadow
              </option>
              <option
                value="grayscale"
              >
                grayscale
              </option>
              <option
                value="resize"
              >
                resize
              </option>
              <option
                value="rotate"
              >
                rotate
              </option>
              <option
                value="sepia"
              >
                sepia
              </option>
              <option
                value="trim"
              >
                trim
              </option>
            </select>
          </div>
          <a
            className="rka-button rka-button-brand rka-button-sm"
            href="#"
            onClick={[Function]}
          >
            Add operation
          </a>
        </div>
      </div>
      <div
        className="col-md-5 col-sm-5"
      >
        <h3
          className="rka-h2 mv-md"
        >
          Preview
          <a
            className="rka-link flo-r txt-sm"
            href="#"
            onClick={[Function]}
          >
            Change picture
          </a>
        </h3>
        <div
          className="rka-stack-img-container bg-chess mb-xs bor-light txt-c"
        >
          <p
            className="pa-md bg-white txt-l"
          >
            Customized 
            <a
              className="rka-link flo-r"
              href="https://test-org.rokka.io/dynamic/noop/7cb0101d53f1579c0e0da11f2407146eebeca638.image/png"
              target="_blank"
            >
              Open in new window
            </a>
          </p>
          <img
            src="https://test-org.rokka.io/dynamic/noop/7cb0101d53f1579c0e0da11f2407146eebeca638.image/png"
          />
        </div>
        <div
          className="rka-stack-img-container bg-chess bor-light txt-c"
        >
          <p
            className="pa-md bg-white txt-l"
          >
            Original 
            <a
              className="rka-link flo-r"
              href="https://test-org.rokka.io/dynamic/noop/7cb0101d53f1579c0e0da11f2407146eebeca638.image/png"
              target="_blank"
            >
              Open in new window
            </a>
          </p>
          <img
            src="https://test-org.rokka.io/dynamic/noop/7cb0101d53f1579c0e0da11f2407146eebeca638.image/png"
          />
        </div>
      </div>
    </div>
  </section>
</form>
`;
